### ISSUE TITLE
Possibility to fail analysis on compiler errors instead of ignoring erroneous files

### ISSUE THREAD

**Issue by selobosi** [SonarSource/SonarTS#862]
**I want to request a feature.**

We were configuring SonarQube with SonarTS and were trying to provoke a Quality Gate failure by creating duplicate code. However, no duplications were detected, whatever we did. We eventually looked closer at the Jenkins console and saw that the SonarQube scanner output the following

```
INFO: Sensor SonarTS [typescript]
INFO: Analyzing 15 typescript file(s) with the following configuration file [...]/tsconfig.json
INFO: 15 files analyzed out of 15
ERROR: Compilation error at [duplicate file 1]:17 "'(' expected."
ERROR: File [duplicate file 1] will be ignored
[more of the above error messages]
ERROR: Compilation error at [duplicate file 2]:17 "'(' expected."
ERROR: File [duplicate file 2] will be ignored
[more of the above error messages]
INFO: Sensor SonarTS [typescript] (done) | time=2557ms
```

I.e. compiler errors lead to files being ignored which lead to a pass of the quality gate. We would like the Jenkins pipeline to fail if there is a compilation error during source code analysis.

Is it already possible to configure SonarTS to achieve this? (I did not find documentation of such a configuration option.) If it is not yet possible to configure analysis to fail on compiler errors: would you consider adding such a feature?

Thanks a lot!

**Comment by selobosi** [comment]
We just figured out that this issue will become drastically less relevant for us when we start compiling the code before the sonarqube analysis ;-). Then compilation itself will fail the pipeline before the sonarqube analysis would ignore the files with compile errors. Please feel free to close this issue.
